app/src/main/java/com/example/datastoreapp/activity/MainActivity.kt

```diff
@@ -16,17 +17,16 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
         viewModel = MainViewModel()
         viewModel.getValue().observe({ lifecycle }, ::updateUI)
         setListeners()
     }
 
     private fun updateUI(data: MainData) {
-        when (data.state) {
-            GO_TO_SHARED_PREFERENCES_VIEW -> {} //start Activity() to shared preferences
-            GO_TO_PROTO_DATASTORE_VIEW -> TODO() //start Activity() to proto datastore
-            GO_TO_PREFERENCES_DATASTORE_VIEW -> TODO() //start Activity() to preferences datastore
-        }
+        this.startActivity(Intent(this, InfoManagerActivity::class.java).apply {
```

Comment:
From the viewModel, you still posting the states GO_TO_SHARED_PREFERENCES_VIEW, GO_TO_PROTO_DATASTORE_VIEW, and GO_TO_PREFERENCES_DATASTORE_VIEW.
I recommend keeping the states here too

Reply:
Also, create a getIntent method into the companion of the InfoManagerActivity that response with the correspondent intent